src/services/dataSource.js

```diff
@@ -1,11 +1,15 @@
-export const getDataSources = (appId) => {
-  const getOptions = appId ? { appId: appId } : {};
+export const getDataSources = appId => {
+  const getOptions = { attributes: 'id,name,accessRules,columns,type' };
```

Comment:
does opening a data source fetch the rest of the required attributes such as settings and hooks?

Reply:
@squallstar  When it opens in the overlay, yes. But as you have mentioned this, the form builder needs hooks from the data source as I recall. I'll check all our integration one more time to ensure that this list is covering all that is we need.

Reply:
Please confirm once you have checked.

Reply:
@squallstar Sorry for the long answer but it seems that there, no need for any additional attributes. When I've tested the provider there were no errors.